theater-gorm/api/theater.go

```diff
@@ -0,0 +1,1345 @@
+	r.HandleFunc("/role", api.getRoleById).Methods("GET")
+	r.HandleFunc("/schedule", api.getScheduleById).Methods("GET")
+	r.HandleFunc("/sector", api.getSectorById).Methods("GET")
+	r.HandleFunc("/ticket", api.getTicketById).Methods("GET")
+	r.HandleFunc("/user", api.getUserById).Methods("GET")
+
+	r.HandleFunc("/account", api.deleteAccountById).Methods("DELETE")
+	r.HandleFunc("/genre", api.deleteGenreById).Methods("DELETE")
+	r.HandleFunc("/hall", api.deleteHallById).Methods("DELETE")
+	r.HandleFunc("/location", api.deleteLocationById).Methods("DELETE")
+	r.HandleFunc("/performance", api.deletePerformanceById).Methods("DELETE")
+	r.HandleFunc("/place", api.deletePlaceById).Methods("DELETE")
+	r.HandleFunc("/poster", api.deletePosterById).Methods("DELETE")
+	r.HandleFunc("/price", api.deletePriceById).Methods("DELETE")
+	r.HandleFunc("/role", api.deleteRoleById).Methods("DELETE")
+	r.HandleFunc("/schedule", api.deleteScheduleById).Methods("DELETE")
+	r.HandleFunc("/sector", api.deleteSectorById).Methods("DELETE")
+	r.HandleFunc("/ticket", api.deleteTicketById).Methods("DELETE")
+	r.HandleFunc("/user", api.deleteUserById).Methods("DELETE")
+
+	r.HandleFunc("/account", api.createAccount).Methods("POST")
+	r.HandleFunc("/genre", api.createGenre).Methods("POST")
+	r.HandleFunc("/hall", api.createHall).Methods("POST")
+	r.HandleFunc("/location", api.createLocation).Methods("POST")
+	r.HandleFunc("/performance", api.createPerformance).Methods("POST")
+	r.HandleFunc("/place", api.createPlace).Methods("POST")
+	r.HandleFunc("/poster", api.createPoster).Methods("POST")
+	r.HandleFunc("/price", api.createPrice).Methods("POST")
+	r.HandleFunc("/role", api.createRole).Methods("POST")
+	r.HandleFunc("/schedule", api.createSchedule).Methods("POST")
+	r.HandleFunc("/sector", api.createSector).Methods("POST")
+	r.HandleFunc("/ticket", api.createTicket).Methods("POST")
+	r.HandleFunc("/user", api.createUser).Methods("POST")
+
+	r.HandleFunc("/account", api.updateAccount).Methods("PUT")
+	r.HandleFunc("/genre", api.updateGenre).Methods("PUT")
+	r.HandleFunc("/hall", api.updateHall).Methods("PUT")
+	r.HandleFunc("/location", api.updateLocation).Methods("PUT")
+	r.HandleFunc("/performance", api.updatePerformance).Methods("PUT")
+	r.HandleFunc("/place", api.updatePlace).Methods("PUT")
+	r.HandleFunc("/poster", api.updatePoster).Methods("PUT")
+	r.HandleFunc("/price", api.updatePrice).Methods("PUT")
+	r.HandleFunc("/role", api.updateRole).Methods("PUT")
+	r.HandleFunc("/schedule", api.updateSchedule).Methods("PUT")
+	r.HandleFunc("/sector", api.updateSector).Methods("PUT")
+	r.HandleFunc("/ticket", api.updateTicket).Methods("PUT")
+	r.HandleFunc("/user", api.updateUser).Methods("PUT")
+}
+
+func (a theaterAPI) getAllTickets(writer http.ResponseWriter, _ *http.Request) {
+	tickets, err := a.data.ReadAllTickets()
+	if err != nil {
+		_, err := writer.Write([]byte("got an error when tried to get tickets"))
+		if err != nil {
+			log.Println(err)
```

Comment:
You just printing an error, and not exit from function, so in step: encoding go model to json you will get an error again.
So please add  `writer.WriteHeader(http.StatusInternalServerError)` and exit of function after printing log message.

Reply:
Done